src/parser.ts

```diff
@@ -440,3 +402,62 @@ function getParamsOrders(ast: AST, tokenIndex: number) {
 
   return paramsOrder;
 }
+
+/**
+ * If the given tag has a default value, then this will add a note to the
+ * description with that default value. This is done because TypeScript does
+ * not display the documented JSDoc default value (e.g. `@param [name="John"]`).
+ *
+ * If the description already contains such a note, it will be updated.
+ */
+function addDefaultValueToDescription(
+  tag: commentParser.Tag,
+): commentParser.Tag {
+  if (tag.optional && tag.default) {
+    let { description } = tag;
+
+    // remove old note
+    description = description.replace(/[ \t]*Default is `.*`\.?$/, "");
+
+    // add a `.` at the end of previous sentences
+    if (description && !/[.\n]$/.test(description)) {
+      description += ".";
+    }
+
+    description += ` Default is \`${tag.default}\``;
+
+    return {
+      ...tag,
+      description: description.trim(),
+    };
+  } else {
+    return tag;
+  }
+}
+
+/**
+ * This will combine the `name`, `optional`, and `default` properties into name
+ * setting the other two to `false` and `undefined` respectively.
+ */
+function combineIntoName({
```

Comment:
IMO, combineIntoName is not good name rename it

Reply:
Suggestions?

Reply:
Also for the future: Please always provide suggestions for renaming. When you see function/variables with strange names, I probably couldn't think of anything better. Suggesting a better name really helps in those situations.

Reply:
assignOptionalAndDefaultToName